File: UI技巧篇/01-UILable文字顶部对齐/01-UILable文字顶部对齐/ViewController.m
//
//  ViewController.m
//  01-UILable文字顶部对齐
//
//  Created by LTH on 2018/11/23.
//  Copyright © 2018 Tony. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    UILabel *testLb = [[UILabel alloc] init];
    NSString *testString = @"My name is Tony Li.And what a awesome day!Today what are you doing now?";
    testLb.text = testString;
    testLb.numberOfLines = 0;
    testLb.backgroundColor = [UIColor lightGrayColor];
    testLb.textAlignment = NSTextAlignmentLeft;
    UIFont *font = [UIFont boldSystemFontOfSize:14.0];
    testLb.font = font;
    NSDictionary *attrs = @{NSFontAttributeName: font};
    // 宽度固定，高度随字体
    CGSize maximumSize = CGSizeMake(300, MAXFLOAT);
    CGRect testSize = [testString boundingRectWithSize:maximumSize options:NSStringDrawingUsesLineFragmentOrigin attributes:attrs context:nil];
    CGRect testFrame = CGRectMake(10, 64 + 10, 300, testSize.size.height);
    testLb.frame = testFrame;
    [self.view addSubview:testLb];
    
}


@end

File: 动画/03-绘图/03-绘图/ViewController.swift
//
//  ViewController.swift
//  03-绘图
//
//  Created by LTH on 2018/10/30.
//  Copyright © 2018 Tony. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        let cgView = CGView.init(frame: CGRect(x: 0,
                                               y: 0,
                                               width: view.frame.width - 20,
                                               height: view.frame.height - 40))
        cgView.backgroundColor = UIColor.purple
        cgView.center = view.center
        view.addSubview(cgView)
    }


}

